### PR TITLE
Adding marketplace-tests maven module, which provides marketplace testing

### PR DESCRIPTION
**Pull request by wyvie**
New module, tests for marketplace. Will probably change some dependencies later, and move some files around, like `CustomWebDriverProvider`, which I had to copy from `ui-tests` for now.

//skip-ci

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: marketplace-tests/pom.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
+		<artifactId>syndesis-qe-parent</artifactId>
+		<groupId>io.syndesis.qe</groupId>
+		<version>0.1-SNAPSHOT</version>
+	</parent>
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>marketplace-tests</artifactId>
```
Comment (avano) [inline marketplace-tests/pom.xml:12]: it is needed to have it as separate module instead of adding it to `ui`?

Comment (avano) [reply marketplace-tests/pom.xml:12]: I understand the intention, but the question is if it is worth for small amount of tests?

File: marketplace-tests/src/test/resources/openshift/quay-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
...YUZOSVVsUnJjbHBzZVhGVE1sbFljbTFxUTJOUmJDMTJVWEYxTWxKMk4wNUNjRzFuTkUxUGRGaEZiSEIzWVRCcVUxcFRObGcwWHpkS1ZURldjblJzTlVGRGExOTBWM1JuVnpCUlJUUlVVR3BGWWpSWFkxUnZibU5LVDB0ZllVRkdkMVJ2WjNoRVp6WlFVWFV5Um1OdFlUZFpUVWx2VWsxVVptTlNiWGxUZVVoYVpVMTJOekJHWm5ZeFEwWlVSR0kxYlRCeGFERlJla3RVVkVGdFdYaFVla3hxY0ZoSlRVdzBRbHBaYzE5U1ptRkVaMHB6WlY5VlJDMVVRVkpMVm5wWE5WWkJSa2hETjBGa2NEQnFUV0ZGV0VJMGRqTlBkR0puV0hKTWNHWTFSek5RYm5CUWVHeE5NbFZ4ZFZNM2JFcEplbVpxUmkxNGFpMXJZbkIwTmkxVmFYZEhUek0wVTBkRmJrZG5XRE50TTBsNVRqVkZWazgwZHpKRlprZ3hXSFJPYlZaQmFtODNUV281VFVGS2IxZHpkSFpoVmpNMlNXTnRkalZUTW5CTGJEQnpjMlV3TjJkVWVucE1ZbFJXWW5kaVdVOXVlbmsyY0V4dVFtNVFaMk5EWTNaWmQzUXpURWhKTmxWbloyRkJWM05ZTUZnM1RrWlZhRFJITTI5WU0xSkJVR1p1YlRKME1VOUdSMEpuV1ROTVV6SlpUVFpKVnpWa1QxVk5hWEJEU25kallrOVFWMW8xZEVVMGFWOXpRMGRrZUVaM04ya3llRU5rVG1wTU16ZFJOWFpUUTJWR2NtbGxhbU5STFc4emFFaHNhMEp4Y2twUk9XNVhObFZFYkRCUVNFNXpWakJLUVZnd2RYUnJTMUZVZEV4U1VrUnRhVkpqUVhGbFVETm5hWEl5Tmw5TGVYbGtVbW8xVVMxVmNFTlJSRU55T1hCdWVXNUhkV1pKY25Sek1uRjNUWGRhY3pNNGFXUmZhMEZhV0V3eVJWaE5ObGR3U21VNVkwdElWbTEyY0VwaVpuaHJWeTFzYmsxc1owVnpUMmxvZEZrMloxVmZhRWRSYVVkeVYybHRSelYzU2xadFRrRmZOM3AzTUZkNlZVeDVNMmRQV2tab1dUYzFSME16VTJRNVkwVllOMHRVYlRBMGNVWllhR2h5V21sMGJ5MHdNREE9IgoJCX0KCX0KfQo=
```
Comment (avano) [inline marketplace-tests/src/test/resources/openshift/quay-secret.yaml:6]: :cry:

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: marketplace-tests/src/test/java/io/syndesis/qe/tar/Compress.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+public class Compress {
```
Comment (avano) [inline marketplace-tests/src/test/java/io/syndesis/qe/tar/Compress.java:22]: wouldn't it be better to use some library for that?

File: marketplace-tests/src/test/java/io/syndesis/qe/util/HelperFunctions.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    public static String encodeFileToBase64Binary(String fileName) throws IOException {
+        File file = new File(fileName);
+        byte[] encoded = Base64.encodeBase64(FileUtils.readFileToByteArray(file));
+        return new String(encoded, StandardCharsets.US_ASCII);
+    }
+
+    public static String readResource(String resourceName) throws IOException {
+        StringBuffer sb = new StringBuffer("");
+        URL url = MarketplaceTest.class.getClassLoader().getResource(resourceName);
```
Comment (avano) [inline marketplace-tests/src/test/java/io/syndesis/qe/util/HelperFunctions.java:87]: use `FileUtils.readFileToString`

File: marketplace-tests/src/test/java/io/syndesis/qe/util/HelperFunctions.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        return executeRequest(httpPost, body, auth);
+    }
+    public static String doDeleteRequest(String url, String body, String auth) throws IOException {
+        HttpDelete httpDelete = new HttpDelete(url);
+
+        return executeRequest(httpDelete, body, auth);
+    }
+
+    private static String executeRequest(HttpUriRequest request, String body, String auth) throws IOException {
+        HttpClient client = HttpClients.createDefault();
```
Comment (avano) [inline marketplace-tests/src/test/java/io/syndesis/qe/util/HelperFunctions.java:112]: you can use `HttpUtils` class from our utitilies

File: marketplace-tests/src/test/java/io/syndesis/qe/MarketplaceTest.java [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    private static final String QUAY_PUSH_URL = "https://quay.io/cnr/api/v1/packages/" + TestConfiguration.quayNamespace();
+    private static final String QUAY_REPOSITORY_URL =  "https://quay.io/api/v1/repository/" +
+        TestConfiguration.quayNamespace() + "/" + "QUAY_PACKAGE";
+    private static final String QUAY_CHANGE_VISIBILITY_URL = QUAY_REPOSITORY_URL + "/changevisibility";
+
+    @BeforeClass
+    public static void setupMarketplace() throws IOException {
+        quayPackage = createQuayProject();
+
+        deployOperatorToOpenshhift(quayPackage);
```
Comment (avano) [inline marketplace-tests/src/test/java/io/syndesis/qe/MarketplaceTest.java:73]: can the "setup" and "teardown" be implemented as `Resource` instance in https://github.com/syndesisio/syndesis-qe/tree/master/utilities/src/main/java/io/syndesis/qe/resource/impl ?

File: utilities/src/main/java/io/syndesis/qe/TestConfiguration.java [hunk 7]
```diff
@@ -96,6 +96,11 @@
 
     private static final String SYNDESIS_APPEND_REPOSITORY = "syndesis.config.append.repository";
 
+    private static final String QUAY_USERNAME = "syndesis.config.quay.username";
+    private static final String QUAY_PASSWORD = "syndesis.config.quay.password";
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/TestConfiguration.java:100]: these two theoretically can be extracted from the pull secret if you want

File: marketplace-tests/src/test/resources/openshift/create-operatorgroup.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: fo-operatorgroup
+  namespace: OPENSHIFT_PROJECT
+spec:
+  targetNamespaces:
+    - OPENSHIFT_PROJECT
```
Comment (avano) [inline marketplace-tests/src/test/resources/openshift/create-operatorgroup.yaml:8]: add newlines at the end of the files

File: marketplace-tests/src/test/java/io/syndesis/qe/MarketplaceTest.java [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+        CustomResourceDefinitionContext subscriptionCrdContext = new CustomResourceDefinitionContext.Builder()
+            .withGroup("operators.coreos.com")
+            .withPlural("subscriptions")
+            .withScope("Namespaced")
+            .withVersion("v1alpha1")
+            .build();
+
+        OpenShiftUtils.getInstance().customResource(subscriptionCrdContext)
+            .createOrReplace(TestConfiguration.openShiftNamespace(),
+                new ByteArrayInputStream(subscriptionYaml.getBytes(StandardCharsets.UTF_8)));
```
Comment (asmigala) [inline marketplace-tests/src/test/java/io/syndesis/qe/MarketplaceTest.java:199]: The `RawCustomResourceOperationsImpl` actually has `public Map<String, Object> createOrReplace(String namespace, String objectAsString)` so no need for the inputstream dance like with regular resources.

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request closed without merging**